### ISSUE TITLE
[Bug] 当使用切片（如 [:, 0:64]）时，会产生 BufferLoad 对象

### ISSUE THREAD

**Issue by liuwenda4** [tile-ai/tilelang-ascend#1265]
## Bug Description
在使用这样类似的切片时，
```
T.tile.exp(
            acc_s_ub[
                compute_side,
                :,
                sub_col : sub_col
                + 64,
            ],
            acc_s_ub[
                compute_side,
                :,
                sub_col : sub_col
                + 64,
            ]
        )
```
会将bufferregion识别为bufferload，从而导致出现error
error: <class 'tvm.tir.expr.BufferLoad'> has no attribute access_ptr，
## Environment

**Hardware:**
- NPU Device: [e.g. Ascend A2, A3]
- Other hardware info: [e.g. CPU, memory]
a2，a3
**Software:**
- OS: [e.g. Ubuntu 20.04, EulerOS]
- CANN version: [e.g. 8.3.RC1]
- torch-npu version: [e.g. 2.6.0.RC1]
- Python version: [e.g. 3.10]
- TileLang-Ascend version: [e.g. commit hash or tag]
- Installation method: [wheel package / build from source]

## Code to Reproduce

```python
# Minimal reproducible code
import tilelang
import torch

# Your code here
```
## Error Message

```
Paste the full error message here
```
error: <class 'tvm.tir.expr.BufferLoad'> has no attribute access_ptr，
## Actual Behavior
A clear and concise description of what actually happened.

## Additional Context

**Programming Mode:**
- [ ] Developer mode (automatic sync/buffer management)
- [ ] Expert mode (manual T.Scope/T.barrier_all)
- [x] Mixed mode

**Backend:**
- [x] Ascend C & PTO backend
- [ ] AscendNPU IR backend

**Pass Configs (if applicable):**
```python
pass_configs = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: ...,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: ...,
    # ...
}
```

**Operator Type:**
[e.g. GEMM, Flash Attention, Softmax, Convolution, etc.]

**Additional Information:**
- Have you checked the [Programming Guide](../docs/TileLang-Ascend%20Programming%20Guide.md)?
- Have you searched existing issues?
- Any other context or screenshots?

**Comment by liggest** [comment]
这里提供两个复现样例：

首先在 [`ascend_tile.py`](https://github.com/tile-ai/tilelang-ascend/blob/ascendc_pto/tilelang/language/ascend_tile.py#L914) 的 `exp` 函数定义内添加如下打印：
```python
def exp(dst: Buffer | BufferRegion, src0: Buffer | BufferRegion):
    print(dst, type(dst))  # <=
    print(src0, type(src0))  # <=
    return unary_op(dst, src0, "exp")
```

例1：
```python
from tilelang import language as T

@T.prim_func
def test():

    with T.Kernel(1, is_npu=True) as (cid, _):
        acc_s_ub = T.alloc_ub([512], "float32")

        T.tile.exp(
            acc_s_ub[:32],
            acc_s_ub[16:32]
        )
```

运行结果：
```ruby
acc_s_ub[0:32] <class 'tvm.tir.stmt.BufferRegion'>
acc_s_ub[16:32] <class 'tvm.tir.expr.BufferLoad'>
error: <class 'tvm.tir.expr.BufferLoad'> has no attribute access_ptr
 --> /.../test_buffer_load.py:10:9
    |  
 10 |          T.tile.exp(
    |          ^          
```

例2：
```python
from tilelang import language as T

def fake_exp(x, y):
    print(x, type(x))
    print(y, type(y))

def debug(x):
    print(x, type(x))
    return x


@T.prim_func
def test():

    with T.Kernel(1, is_npu=True) as (cid, _):
        acc_s_ub = T.alloc_ub([2, 16, 512], "float32")
        compute_side = T.Var("compute_side", "int32")
        sub_col = T.Var("sub_col", "int32")

        fake_exp(acc_s_ub[
            compute_side,
            :,
            sub_col : sub_col
            + 64,
        ], acc_s_ub[
            compute_side,
            :,
            sub_col : sub_col
            + 64,
        ])

        T.tile.exp(
            debug(acc_s_ub[
                compute_side,
                :,
                sub_col : sub_col
                + 64,
            ]),
            acc_s_ub[
                compute_side,
                :,
                sub_col : sub_col
                + 64,
            ]
        )
```

运行结果：
```ruby
acc_s_ub[compute_side, 0:16, sub_col:sub_col + 64] <class 'tvm.tir.stmt.BufferRegion'>
acc_s_ub[compute_side, 0:16, sub_col:sub_col + 64] <class 'tvm.tir.stmt.BufferRegion'>
acc_s_ub[compute_side, 0:16, sub_col:sub_col + 64] <class 'tvm.tir.stmt.BufferRegion'>
acc_s_ub[compute_side, 0:16, sub_col:sub_col + 64] <class 'tvm.tir.stmt.BufferRegion'>
acc_s_ub[compute_side, 0:16, sub_col:sub_col + 64] <class 'tvm.tir.expr.BufferLoad'>
error: <class 'tvm.tir.expr.BufferLoad'> has no attribute access_ptr
 --> /.../test_buffer_load.py:36:9
    |  
 36 |          T.tile.exp(
    |          ^            
```

**Comment by liggest** [comment]
这种 Buffer 切片在特殊条件下不转成 BufferRegion 而是转成 BufferLoad 的现象，疑似是当前框架依赖的 TVM 版本的一种行为

AI 定位出的触发条件如下：
- 被切片的 Buffer 作为参数传入函数
- 调用的函数属于 “属性调用” （`obj.func(...)` 的形式，例如 `T.tile.xxx(...)`），且函数名不是 `reads`、`writes`、`match_buffer`、`realize`、`copy`
> 详见 [3rdparty/tvm/python/tvm/script/parser/core/evaluator.py:177-181](https://github.com/tile-ai/tvm/blob/c2921fdaf795b1103d21abc962e83a209c7258d7/python/tvm/script/parser/core/evaluator.py#L177-L181)
- 切片的形式为 `buffer[lower : upper]`，`lower` 与 `upper` 必须都存在，会自动注入步长 `step=1`
> 详见 [3rdparty/tvm/python/tvm/script/parser/core/evaluator.py:202-213](https://github.com/tile-ai/tvm/blob/c2921fdaf795b1103d21abc962e83a209c7258d7/python/tvm/script/parser/core/evaluator.py#L202-L213)
-  带步长的 Buffer 切片被转换成 BufferLoad，无步长则为普通 BufferRegion
> 详见 [3rdparty/tvm/python/tvm/tir/buffer.py:187-221](https://github.com/tile-ai/tvm/blob/c2921fdaf795b1103d21abc962e83a209c7258d7/python/tvm/tir/buffer.py#L187-L221)

因此，上述复现样例中：
```python
T.tile.exp(
    acc_s_ub[:32],
    acc_s_ub[16:32]
)
```
- 第一个 `acc_s_ub[:32]` 切片没有 `lower`，只有 `upper=32`，未触发条件
- 第二个 `acc_s_ub[16:32]` 切片 `lower=16` 且 `upper=32`，由打印输出得知，被转换成了 BufferLoad

```python
fake_exp(acc_s_ub[
    compute_side,
    :,
    sub_col : sub_col
    + 64,
], acc_s_ub[
    compute_side,
    :,
    sub_col : sub_col
    + 64,
])

T.tile.exp(
    debug(acc_s_ub[
        compute_side,
        :,
        sub_col : sub_col
        + 64,
    ]),
    acc_s_ub[
        compute_side,
        :,
        sub_col : sub_col
        + 64,
    ]
)
```

- `fake_exp(...)` 不属于 `T.xxx` 形式的 “属性调用”，未触发条件
- `T.tile.exp(...)` 属于 “属性调用”，但 `debug(...)` 函数不属于，故 `T.tile.exp(...)` 的第一个参数未触发条件，而第二个参数触发


```python
def unary_op(dst: Buffer | BufferRegion, src0: Buffer | BufferRegion, op: str):
    if isinstance(dst, BufferRegion):
        dst_ptr, dst_extent = _handle_buffer_region(dst, "w")
    else:
        dst_ptr = dst.access_ptr("w")
        dst_extent = dst.shape
    if isinstance(src0, BufferRegion):
        src0_ptr, src0_extent = _handle_buffer_region(src0, "r")
    else:
        src0_ptr = src0.access_ptr("r")
        src0_extent = src0.shape
    ...
```

当前 `T.tile.exp(...)` 内部的 `unary_op(...)`，只支持处理 Buffer 和 BufferRegion，不支持 BufferLoad，传入 BufferLoad 后因其没有 `access_ptr` 函数，故产生错误：`error: <class 'tvm.tir.expr.BufferLoad'> has no attribute access_ptr`

**Comment by liggest** [comment]
看起来 #1207 中问题的前半部分也与这个有关

**Comment by liggest** [comment]
目前来看，这种 TVM 行为从算子、框架层面，大概有如下几个规避或解决手段：

- 从 `ascend_tile.py` 直接导入接口函数，避免 “属性调用”
```python
from tilelang.language.ascend_tile import exp

exp(...)
```
- 定义无操作透传函数，避免 “属性调用”
```python
def just(x):
    return x

T.tile.exp(
    just(...),
    just(...)
)
```

- 为各个只支持 BufferRegion 的接口都额外支持传入 BufferLoad，使用通用的逻辑把传入的 BufferLoad 转成 BufferRegion
可以参考 `T.copy` 和 `T.atomic_add` 内的逻辑，看看有没有能复用的部分
- 升级 TVM 版本
如 #1207 中所述，在 [[TIRx] Bringup TIRx Infrastructure (apache/tvm@859498d)](https://github.com/apache/tvm/commit/859498dc01660d06f61d54c98edf27828d68f25e#diff-89ddf286895419979dea0e747b22ca611aed7f350650722eb9c4571c88ffd818) 这个 commit 之后，TVM 中判断 Buffer 切片是否带步长的逻辑有修改，避免了这种特殊条件下转成 BufferLoad 行为的发生

**Comment by liggest** [comment]
https://github.com/tile-ai/tilelang-ascend/pull/1295 看起来解决了这个问题，需要确认一下
